implementations/rust/ockam/ockam_node/src/relay/worker_relay.rs

```diff
@@ -136,13 +136,17 @@ where
                         Ok(false) => {
                             break;
                         },
-                        // An error occured -- log and continue
+                        // An error occurred -- log and continue
                         Err(e) => error!("Error encountered during '{}' message handling: {}", address, e),
                     }
                 },
-                _ = ctrl_rx.recv() => {
-                    debug!("Relay received shutdown signal, terminating!");
-                    break;
+                result = ctrl_rx.recv() => {
+                    if result.is_some() {
+                        debug!("Relay received shutdown signal, terminating!");
+                        break;
+                    }
```

Comment:
I assume this gates the `break` condition from the case where the control channel is dropped and returns `None`?

Reply:
@spacekookie yes, I don't like that solution, there is race condition described in #2283. Would be great to solve that properly. For now I just thought reasonable to distinguish between dropping a tx side and receiving a signal, because at least debug message was misleading - it printed that we got shutdown signal even if we didn't